### DISCUSSION TITLE
New markdown-exec version: updating ToC with generated headings

### DISCUSSION THREAD

**Discussion by pawamoy** [pdm-project/pdm#1834] [Show and tell]
Hey @frostming!

I've just released Markdown Exec 1.5.0 which now updates the ToC with the generated headings, by default.
If you update it and rebuild your docs, you should see the populated ToC sidebar for the CLI reference :tada: 

If you have code blocks for which you don't want to update the ToC, use the `updatetoc="no"` option.

(*this feature was incredibly hard to implement :sweat_smile: python-markdown is not an easy beast to handle*)

**Comment by frostming** [comment]
> python-markdown is not an easy beast to handle

Consider another markdown parser that is easier to extend?

https://github.com/frostming/marko

**Reply by pawamoy** [reply]
I'd love to try, but I'm maintaining this extension for it to be used within MkDocs, which entirely relies on Python-Markdown.

**Reply by frostming** [reply]
I see. BTW the docs are updated: https://pdm.fming.dev/dev

**Reply by pawamoy** [reply]
Nice! Let me know if you'd like other features like the ability to choose the id prefix (instead of `exec-N--`).

**Reply by frostming** [reply]
The prefix doesn't matter, just that the N is unstable across rebuild and reloading.

**Reply by pawamoy** [reply]
Yeah, I'm yet to find a way to handle live-reloading in a stable way. Maybe there's a way to reuse the `id` option here, since it's supposed to be unique across all the pages. You'd then just have to add an id, like 

````
```python exec="true" id="cli"
````

...and the generated ids for headings would be `#cli-list` instead of `#exec-0--list`. WDYT?

**Reply by pawamoy** [reply]
Oh well, just found a one-line addition that resets the counter to keep it stable across rebuilds and live-reload :joy:

**Reply by pawamoy** [reply]
v1.5.2 released :slightly_smiling_face: Thanks for the feedback!

**Reply by pawamoy** [reply]
v1.6.0 released with the [`idprefix` option](https://pawamoy.github.io/markdown-exec/usage/#html-ids) :slightly_smiling_face:
